yarn/src/test/scala/org/apache/spark/deploy/yarn/BaseYarnClusterSuite.scala

```diff
@@ -144,7 +146,12 @@ abstract class BaseYarnClusterSuite
 
     extraConf.foreach { case (k, v) => props.setProperty(k, v) }
 
-    val propsFile = File.createTempFile("spark", ".properties", tempDir)
+    val submitConf = extraConf ++ Map(
+      // SPARK-4267: make sure java options are propagated correctly.
+      "spark.driver.extraJavaOptions" -> "-Dfoo=\"one two three\"",
+      "spark.executor.extraJavaOptions" -> "-Dfoo=\"one two three\"")
+
+    val propsFile = createConfFile(childClasspath = childClasspath, extraConf = submitConf)
     val writer = new OutputStreamWriter(new FileOutputStream(propsFile), UTF_8)
     props.store(writer, "Spark properties.")
```

Comment:
I'm confused -- those props related to SPARK-4267 are already set above on line 132, and then they are written to the file here.  It looks like you are ignoring `props` when initially writing the file w/ `createConfFile`, but then overwriting it again with this call to `props.store`.